README.md

```diff
@@ -203,51 +205,65 @@ Open the project in your IDE to configure the code.
-token_acquisition_result = msal_instance.acquire_token_by_authorization_code(authorization_code, config.get('SCOPES'))
-```
+In this sample, much of the required MSAL for Python configurations are automatically setup using utilties found in [Microsoft Identity Python Samples Common](https://github.com/azure-samples/ms-identity-python-common). For a more direct, hands-on demonstration of the sign-in process without this abstraction, please see the code within this [Python Webapp](https://github.com/azure-samples/ms-identity-python-webapp) sample.
 
-MSAL Python:
-* Downloads the Azure AD metadata, including signing keys, and finds the issuer name for the tenant.
-* Processes Azure AD responses by validating the signature and issuer in the ID and/or Access tokens.
-* Parses the ID Token claims into plaintext.
- 
-The result is then put into a server-side session, in the following properties:
-```python
-session['msal_token_result']=token_acquisition_result
-session['msal_authenticated']=True
-id_token_claims=token_acquisition_result.get('id_token_claims', {})
-session['msal_id_token_claims'] = id_token_claims
-session['msal_username'] = id_token_claims.get('name', None)
-```
+At a minimum, following parameters need to be provided to MSAL for Python:
+  - The **Client ID** of the app
+  - The **Client Credential**, which is a requirement for Confidential Client Applications
+  - The **Azure AD Authority**, which includes the Tenant ID of the AAD application in this sample's scenario
+
+1. The first step of the sign-in process is to send a request to the `/authorize` endpoint on Azure Active Directory.
+
+1. An MSAL for Python **ConfidentialClientApplication** instance is created:
+
+    ```python
+    client_instance = msal.ConfidentialClientApplication(
+      client_id=CLIENT_ID,
+      client_credential=CLIENT_CREDENTIAL,
+      authority=AUTHORITY,
+    )
+    ```
+
+1. The `client_instance` instance is leveraged to construct a `/authorize` request URL with the appropriate parameters, and the browser is redirected to this URL.
+1. The user is presented with a sign-in prompt by Azure Active Directory. If the sign-in attempt is successful, the user's browser is redirected back to this app's `/redirect` endpoint. A successful request to this endpoint will contain an **authorization code**.
+1. The `client_instance` is used to exchange this authorization code for an ID Token and Access Token from Azure Active Directory.
+
+    ```python
+    token_acquisition_result = client_instance.acquire_token_by_authorization_code(authorization_code, SCOPES)
+    # this sends the authorization code to Azure AD's `/token` endpoint to request a token.
+    ```
+
+1. If the request is successful, MSAL for Python validates the signature and nonce of the incoming token. If these checks succeed, it returns the resulting `id_token`, `access_token` and plaintext `id_token_claims` in a dictionary. *It is the application's responsibility to store these tokens securely.*
```

Comment:
>`access_token` and plaintext [](start = 168, length = 28)

does this sample gets access_token

Reply:
yes, the access token is obtained by msal library when auth code flow is used.